src/ManifoldMuseum.jl

```diff
@@ -3,24 +3,91 @@ module ManifoldMuseum
 import Base: isapprox, exp, log
 import LinearAlgebra: dot, norm
 
+export Manifold
+export dimension,
+    distance,
+    dot,
+    exp!,
+    geodesic,
+    log!,
+    norm,
+    retract,
+    retract!,
+    typical_distance,
+    zero_tangent_vector,
+    zero_tangent_vector!
 
 abstract type Manifold end
 
-isapprox(m::Manifold, x, y; kwargs...) = isapprox(x, y; kwargs...)
+"""
+    isapprox(M::Manifold, x, y; kwargs...)
+
+Check if points `x` and `y` from manifold `M` are approximately equal.
+
+Keyword arguments can be used to specify tolerances.
+"""
+isapprox(M::Manifold, x, y; kwargs...) = isapprox(x, y; kwargs...)
+
+"""
+    isapprox(M::Manifold, x, v, w; kwargs...)
+
+Check if vectors `v` and `w` tangent at `x` from manifold `M` are
+approximately equal.
+
+Keyword arguments can be used to specify tolerances.
+"""
+isapprox(M::Manifold, x, v, w; kwargs...) = isapprox(v, w; kwargs...)
+
+"""
+    retract!(M::Manifold, y, x, v, t=1)
+
+Retraction (cheaper, approximate version of exponential map) of tangent
+vector `t*v` at point `x` from manifold `M`.
+Result is saved to `y`.
+"""
+retract!(M::Manifold, y, x, v, t=1) = exp!(M, y, x, v, t)
+
+"""
+    retract(M::Manifold, x, v, t=1)
 
-retract!(M::Manifold, y, x, v) = exp!(M, y, x, v)
-retract(M::Manifold, x, v) = retract!(M, copy(x), x, v)
+Retraction (cheaper, approximate version of exponential map) of tangent
+vector `t*v` at point `x` from manifold `M`.
+"""
+retract(M::Manifold, x, v, t=1) = retract!(M, copy(x), x, v, t)
 
 project_tangent!(M::Manifold, w, x, v) = error("Not implemented")
 project_tangent(M::Manifold, x, v) = project_tangent!(M, copy(x), x, v)
 
 distance(M::Manifold, x, y) = error("Not implemented")
```

Comment:
Since distance is defined by the inner product, couldn't we have a default implementation of
```julia
distance(M::Manifold, x, y) = norm(M, x, log(M, x, y))
```

Of course it could be specialized for certain manifolds.

Reply:
Yes, I think that is a good default. If someone has a faster method for his/her manifold, you can really still specialise, but If you just want to get started, its a function less to implement.